guacamole-common-js/src/main/webapp/modules/AudioRecorder.js

```diff
@@ -183,14 +183,19 @@ Guacamole.RawAudioRecorder = function RawAudioRecorder(stream, mimetype) {
 
     /**
      * A function which directly invokes the browser's implementation of
-     * navigator.getUserMedia() with all provided parameters.
+     * navigator.mediaDevices.getUserMedia() with all provided parameters.
      *
      * @type Function
      */
-    var getUserMedia = (navigator.getUserMedia
-            || navigator.webkitGetUserMedia
-            || navigator.mozGetUserMedia
-            || navigator.msGetUserMedia).bind(navigator);
+
+    if (navigator.mediaDevices === undefined)
+        navigator.mediaDevices = {};
+
+    if (navigator.mediaDevices.getUserMedia === undefined)
```

Comment:
Since this will only be valid if `navigator.mediaDevices` was not defined in the first place (we'd need to `bind()` to `navigator` otherwise). I'd suggest:

```javascript
navigator.mediaDevices = {

    /**
     * That comment from before.
     */
    getUserMedia: (...all those tests...).bind(navigator)

};
```

Reply:
So, the reason I went this route is based on one of the examples on that MDN page ([1]).  The comment in the code at the bottom of the page says:

    // Some browsers partially implement mediaDevices. We can't just assign an object
    // with getUserMedia as it would overwrite existing properties.
    // Here, we will just add the getUserMedia property if it's missing.

Thus, I was going for the same level of compatibility as defined in that code sample.

Reply:
Ah, OK. That makes good sense.